src/ui/public/doc_table/components/table_row.js

```diff
@@ -77,6 +78,35 @@ module.directive('kbnTableRow', function ($compile) {
 
         $detailsScope.row = $scope.row;
 
+        $detailsScope.copyTextToClipboard = text => {
```

Comment:
There's an existing clipboard copy implementation in the codebase already: https://github.com/elastic/kibana/blob/a985840692abce8e7a5184e5175e2d294853513a/src/ui/public/share/directives/share.js#L147-L147

Let's extract that existing logic into a service that both the share UI and this doc link UI can leverage.